### ISSUE TITLE
Not able to download more than 250 records

### ISSUE THREAD

**Issue by vg15** [mobz/elasticsearch-head#224]
I am using ES1.7.3 and I am not able to download more than 250 records in the CSV file .

Please help .

**Comment by philipskokoh** [comment]
Do you see any error messages? It's working fine when I tested it.

**Comment by vg15** [comment]
No Error ,It starts processing but then nothing appears. Let me tell you what's happening. Query is giving me 3500 records and when I select No of records from the drop down i.e 5000 I can see "Generate Download link" .Now after clicking it I can see new link "Download' appears . Here is the problem Save as dialog box only appears if I select 250 or less records .It is not appearing when I select more than 250.

Is it a browser issue ? OR ES1.7.3 issue ?

**Comment by philipskokoh** [comment]
I don't experience the problem here.. But if I set the number of returned results too high (for example 25000), I get more chance to get timeout. I think this is what you experienced.

**Comment by vg15** [comment]
Strange I am not sure why I am experiencing this problem . Can you tell me why I am not able to see anything on the structured query tab . It is blank , even Indices tab is not getting opened .I was not facing these issues in older version of head .

**Comment by philipskokoh** [comment]
It should work, not sure what happened to your elasticsearch head. It try on ES 1.7.0, and it's working fine.

Did you remove the old head before installing the new version? I suggest you remove the older version first before installing new version.
https://www.elastic.co/guide/en/elasticsearch/plugins/2.1/listing-removing.html

**Comment by philipskokoh** [comment]
@vg15: Check the latest update on my branch #218. It should solve your problem.
Reference:
http://stackoverflow.com/questions/23301467/javascript-exporting-large-text-csv-file-crashes-google-chrome

**Comment by vg15** [comment]
@Philips,

Thanks For the response , so which branch do I need do download ? Please
let me know the exact path or location ?

On Tue, Dec 15, 2015 at 7:03 AM, Philips Kokoh <notifications@github.com>
wrote:

> @vg15 <https://github.com/vg15>: Check the latest update on my branch #218
> <https://github.com/mobz/elasticsearch-head/pull/218>. It should solve
> your problem.
> Reference:
>
> http://stackoverflow.com/questions/23301467/javascript-exporting-large-text-csv-file-crashes-google-chrome
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mobz/elasticsearch-head/issues/224#issuecomment-164613264>
> .
>



-- 
*Regards,*
*Vikas Gopal*

**Comment by philipskokoh** [comment]
It depends on your document size.
If your documents are big, it may take a while. In my server, it takes about few minutes and up to few GB to retreive 25000 big documents. For small data documents, 25000 records are fast.

**Comment by vg15** [comment]
Thanks again Philips Actually I waited for 30 min , infact it is still
searching . I just have 5600 documents , I am not getting it what does
document size means ? Do we have any way to reduce document size ?

On Tue, Dec 15, 2015 at 2:44 PM, Philips Kokoh <notifications@github.com>
wrote:

> It depends on your document size.
> If your documents are big, it may take a while. In my server, it takes
> about few minutes and up to few GB to retreive 25000 big documents. For
> small data documents, 25000 records are fast.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mobz/elasticsearch-head/issues/224#issuecomment-164695144>
> .
>



-- 
*Regards,*
*Vikas Gopal*

**Comment by philipskokoh** [comment]
30 mins is too long. You probably get timeout...
Can you try again when the cluster health is green?

**Comment by vg15** [comment]
Yeah I should have try this with Status green , let me start some ES
instances to make it green , will update you .Thanks again.

On Tue, Dec 15, 2015 at 3:30 PM, Philips Kokoh <notifications@github.com>
wrote:

> 30 mins is too long. You probably get timeout...
> Can you try again when the cluster health is green?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mobz/elasticsearch-head/issues/224#issuecomment-164708344>
> .
>



-- 
*Regards,*
*Vikas Gopal*

**Comment by vg15** [comment]
No ! I have started 3 ES instances to make cluster health Green , even after doing this I can only see a blank page as I posted above . I waited for 5-10 min , is it an environment issue I am on windows 8 , using Chrome web browser ?

**Comment by philipskokoh** [comment]
Hmm, not sure what happened.. I don't have windows 8 so I cannot test it. But chrome should work the same in any OS.
Can you do these:
1. run the same query, but use other output such as json? It's to test whether the problem is in query part or in csv generation part. I feel something is not right with the query, not the csv output.
2. open 'Developer tools' in your chrome, then go to console tab? You can copy any error messages in the console.

**Comment by vg15** [comment]
So one thing is confirmed that it is not an environment issue as issue is same for Windows Server 2012 R2.Well below is the error I got in the development tool while selecting 25000 for JSON or for CSV.

![image](https://cloud.githubusercontent.com/assets/11420150/11838691/b096902a-a40e-11e5-9b81-69c838ba4f9c.png)


So the error says 

POST http://localhost:9200/latest/_search 500 (Internal Server Error)jQuery.ajaxTransport.s.send @ vendor.js:7829jQuery.extend.ajax @ vendor.js:7307services.Cluster.ux.Class.extend.request @ app.js:1266services.Cluster.ux.Class.extend.post @ app.js:1277ui.FilterBrowser.ui.AbstractWidget.extend._search_handler @ app.js:3899jQuery.event.handle @ vendor.js:2926jQuery.event.add.elemData.handle.eventHandle @ vendor.js:2560
app.js:1271 Object {XHR Error: "error", message: "Internal Server Error"}

**Comment by philipskokoh** [comment]
Does it also happen when you select json or table as the output?
Looks like the error is in the query part, not in csv generation. Can you share your query? You can check 'show query source'

**Comment by vg15** [comment]
Yes it happens for all the options that is JSON,Table and CSV. I am gettign same error for all the options .
My query is like 

{
"query": {
"bool": {
"must": [
{
"match_all": { }
}
],
"must_not": [ ],
"should": [ ]
}
},
"from": 0,
"size": 25000,
"sort": [ ],
"aggs": { }
}

![image](https://cloud.githubusercontent.com/assets/11420150/11842103/76761636-a427-11e5-97df-3385b1e5da4e.png)

**Comment by philipskokoh** [comment]
I can replicate this error when I retrieve big documents (total size ~300MB). The size is too big to be loaded in a browser. If you want to export big chunk of your ES index (hundreds MB or few GB), I suggest you to explore many available solutions such as 
http://stackoverflow.com/questions/18892560/is-there-any-way-in-elasticsearch-to-get-results-as-csv-file-in-curl-api
